### PR TITLE
User/mjh/om4 05 preprocessing

### PR DESCRIPTION
**Pull request by MJHarrison-GFDL**
References the PHC version with metadata updates. Also added the topography variance to ocean_topog.

### CODE REVIEW

File: ice_ocean_SIS2/OM4_05/preprocessing/md5sums.txt [hunk 1]
```diff
@@ -12,6 +12,7 @@ de35470a7c769ec5351bc1264419db65  mercator_topog_gebco.nc
 9d21a361543ecb173eba26587aa34503  ncap_topog_gebco.nc
 bf6b717361d46d5b3e754bcf374e4b58  ncap_topog.nc
 bdd44ddfb61f5bd304f43f27c1eaf3c0  ocean_topog.nc
+29d8ea0fd410d2214b49e6ffcacac973  ocean_topog_test.nc
```
Comment (adcroft) [inline ice_ocean_SIS2/OM4_05/preprocessing/md5sums.txt]: New file in md5sums?

File: ice_ocean_SIS2/OM4_05/preprocessing/md5sums.txt [hunk 2]
```diff
@@ -12,6 +12,7 @@ de35470a7c769ec5351bc1264419db65  mercator_topog_gebco.nc
 9d21a361543ecb173eba26587aa34503  ncap_topog_gebco.nc
 bf6b717361d46d5b3e754bcf374e4b58  ncap_topog.nc
 bdd44ddfb61f5bd304f43f27c1eaf3c0  ocean_topog.nc
```
Comment (adcroft) [inline ice_ocean_SIS2/OM4_05/preprocessing/md5sums.txt]: Shouldn't ocean_topog.nc md5sum have changed?

**Comment by jkrasting** [comment]
There is an issue with the time axis in the IAF runoff file:

```
FATAL from PE    26: time_interp_external 2: time 710655 (19480101.000000
is before range of list 710669-732539(19480115.000000 -
20071216.000000),file=./INPUT/runoff.daitren.iaf.v2011.02.10.720x576.nc
,field=runoff
```

Also, should this file have a modulo time axis?

**Comment by MJHarrison-GFDL** [comment]
Yes. The IAF and CLIM runoff files need a modulo attribute. In the spirit of cleaniing up things, I prefer to add the attribute to the source files

/archive/gold/datasets/CORE/NYF_v2.0/runoff.daitren.clim.v2011.02.10.nc
/archive/gold/datasets/CORE/IAF_v2.0/runoff.daitren.iaf.v2011.02.10.nc

**Comment by adcroft** [comment]
If I add modulo to the existing files (that have been used successfully
since 2015) that would almost certainly break existing/past runs. Why does
these files work for existing runs and not the new run?

**Comment by MJHarrison-GFDL** [comment]
I can't say why the past runs worked, only that FMS needs the modulo attribute and this has been the case for a number of years. So, I am certain that any successful runs were using files with a modulo attribute. For example, this may have been added at runtime.  We could just make a "corrected" version and I will point to that in the Jenkins script.

**Comment by jkrasting** [comment]
The past runs worked because we were using versions of the files that we
created by Amy.

**Comment by jkrasting** [comment]
I finally got the model to run, but there are still more fixes that I needed to do.  @MJHarrison-GFDL - could you add these to the Makefile?

```
ncks --fix_rec_dmn nEdits ocean_topog.nc ocean_topog_test.nc
```

```
ncatted -O -a modulo_beg,Time,a,c,"1948-01-01 00:00:00" runoff.daitren.iaf.v2011.02.10.720x576.nc
ncatted -O -a modulo_end,Time,a,c,"2008-01-01 00:00:00" runoff.daitren.iaf.v2011.02.10.720x576.nc
```

```
ncatted -O -a units,time,o,c,days since 0001-01-01 00:00:00 salt_restore.nc
ncatted -O -a modulo,time,a,c,  salt_restore.nc
```

**Pull request merged**